src/shared/Core/ITrace2Writer.cs

```diff
@@ -0,0 +1,10 @@
+using System;
+
+namespace GitCredentialManager;
+
+public interface ITrace2Writer : IDisposable
+{
+    public bool Failed { get; }
+
+    public void Write(Trace2Message message);
```

Comment:
No need for the `public` access modifier in an `interface`.

```suggestion
   bool Failed { get; }

   void Write(Trace2Message message);
```